ocaml/xapi/xapi_host.ml

```diff
@@ -1361,48 +1363,46 @@ let certificate_sync ~__context ~host = Certificates.local_sync ()
 let get_server_certificate ~__context ~host =
   Certificates.get_server_certificate ()
 
+let with_cert_lock : (unit -> 'a) -> 'a =
+  let cert_m = Mutex.create () in
```

Comment:
Shouldn't `cert_m` be bound outside of `with_cert_lock` to ensure it's shared in all calls of `with_cert_lock`?

Reply:
This makes a closure, so it should be shared. This is similar, but you can easily see that it works in utop

```
let with_ctr : unit -> unit =
 let r = ref 0 in
 fun () -> Printf.printf "%i" !r; r := !r + 1
```